### PR TITLE
Add `update_time` field to Dispatch message

### LINKED ISSUE #21: Add `update_time` field to Dispatch message
### What's needed?

Update time is required to keep track of when a dispatch was last updated, mainly for caching purposes. It should be updated whenever an update request message touches on a dispatch, and set to the same timestamp as `create_time` when the dispatch is created.

### Proposed solution

_No response_

### Use cases

_No response_

### Alternatives and workarounds

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by stefan-brus-frequenz**
Fixes #21

### CODE REVIEW

**Review by tiyash-basu-frequenz — DISMISSED** [review]

File: proto/frequenz/api/dispatch/dispatch.proto [hunk 1]
```diff
@@ -65,17 +65,22 @@ message Dispatch {
...
-  google.protobuf.Struct payload = 10;
+  google.protobuf.Struct payload = 9;
+
+  // The creation time
+  // This is set when a dispatch is created via the create request message
+  google.protobuf.Timestamp create_time = 10;
+
+  // The update time
+  // This is set when a dispatch is modified via the update request message
+  google.protobuf.Timestamp update_time = 11;
```
Comment (thomas-nicolai-frequenz) [inline proto/frequenz/api/dispatch/dispatch.proto]: I suggest to insert them as 4 and 5 instead.

File: proto/frequenz/api/dispatch/dispatch.proto [hunk 2]
```diff
@@ -189,15 +194,12 @@ message DispatchUpdateRequest {
   // The component selector
   DispatchComponentSelector selector = 6;
```
Comment (thomas-nicolai-frequenz) [inline proto/frequenz/api/dispatch/dispatch.proto]: uh `optional string type = 3;` should be removed. Changing that would create quite some issues.

Comment (thomas-nicolai-frequenz) [reply proto/frequenz/api/dispatch/dispatch.proto]: Also keep in mind that start_time has to be in the future! Can you please also add that to the documentation? Only start_time with a future date should be allowed to be updated.

File: proto/frequenz/api/dispatch/dispatch.proto [hunk 3]
```diff
@@ -151,6 +156,7 @@ message DispatchCreateRequest {
   string type = 2;
 
   // The start time
+  // The starting timestamp has to be in the future when creating a dispatch
```
Comment (thomas-nicolai-frequenz) [inline proto/frequenz/api/dispatch/dispatch.proto]: I'm not 100% certain we should be defining it like that. What if someone wants to dispatch something NOW?

File: proto/frequenz/api/dispatch/dispatch.proto [hunk 4]
```diff
@@ -177,27 +183,22 @@ message DispatchUpdateRequest {
   // The microgrid identifier
   optional uint64 microgrid_id = 2;
 
-  // The type of dispatch
-  optional string type = 3;
-
   // The start time
-  google.protobuf.Timestamp start_time = 4;
+  // The starting timestamp has to be in the future when updating a dispatch
```
Comment (thomas-nicolai-frequenz) [inline proto/frequenz/api/dispatch/dispatch.proto]: Same here. We might want to support NOW as the smallest possible value.

**Review by thomas-nicolai-frequenz — CHANGES_REQUESTED** [review]
The NOW topic should be solved in case someone wants to dispatch something immediately in an emergency situation.

**Review by thomas-nicolai-frequenz — APPROVED** [review]

**Review by tiyash-basu-frequenz — APPROVED** [review]

**Pull request merged**